src/components/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,838 @@
+        emit('input', state.selectedRange)
+        emit('change', state.selectedRange)
+      } else {
+        emit('input', new Date(state.selectedRange.start))
+        emit('change', new Date(state.selectedRange.start))
+      }
+
+      handleClose()
+      updateDisplay()
+    }
+
+    /**
+     * Updates the input field value as a visual confirmation after a choice is made
+     *
+     * If a time range (custom or relative) determine which tab has focus,
+     * then update input field text.
+     * Else, update input field text for single date / time instance
+     */
+    const updateDisplay = () => {
+      if (props.range && hasTimePeriods.value && !showCalendar.value) {
+        state.abbreviatedDisplay = state.selectedTimeframe.display
+      } else {
+        state.abbreviatedDisplay = formatDisplayDate(state.selectedRange, true)
+      }
+    }
+
+    const ucWord = (val: string) => {
+      return val.charAt(0).toUpperCase() + val.slice(1)
+    }
+
+    const handleClose = async () => {
+      state.hidePopover = true
+      await nextTick()
+      state.hidePopover = false
+    }
+
+    /**
+     * Saves the internal state (range or single value) whenever
+     * the `v-calendar` instance is interacted with.
+     */
+    watch(selectedCalendarRange, (newValue, oldValue) => {
+      if (newValue !== undefined && newValue !== oldValue) {
+        changeCalendarRange(newValue)
+      }
+    }, { immediate: true })
+
+    onMounted(() => {
+      // Select the tab based on incoming defaults; save the default value to our internal
+      // state and update the input field to display the human-readable date/time.
+      if (props.modelValue) {
+        if (props.modelValue instanceof Date || !props.modelValue.timePeriodsKey) {
+          state.tabName = 'custom'
+          changeCalendarRange(props.modelValue)
+          updateDisplay()
+        } else {
+          state.tabName = 'relative'
+          for (const section of props.timePeriods) {
+            const selectedTimeframe = section.values.find(e => e.key === props.modelValue.timePeriodsKey)
+            if (selectedTimeframe) {
+              changeRelativeTimeframe(selectedTimeframe)
+              updateDisplay()
+              break
+            }
+          }
+        }
+      }
+    })
+
+    return {
+      changeRelativeTimeframe,
+      clearSelection,
+      handleClose,
+      hasCalendar,
+      hasTimePeriods,
+      modelConfig,
+      selectedCalendarRange,
+      calendarSelectAttributes,
+      calendarDragAttributes,
+      showCalendar,
+      ...toRefs(state),
+      submitDisabled,
+      widthStyle,
+      submitTimeFrame,
+      ucWord,
+    }
+  },
+})
+</script>
+
+<style lang="scss">
+@import '@/styles/variables';
+@import '@/styles/functions';
+
+$timepicker-min-width: 360px;
+$margin: 6px;
+
+.k-datetime-picker {
+  max-width: 100%; // Prevent overflowing the container
+  // For aesthetic purposes when relative time frames are present
```

Comment:
super-nit: can you add some empty lines between the SCSS rules in the style block so it's just easier to read?